### PR TITLE
ci: migrete ut to verifyCI

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 

we want to replace `idc-unit-test` with `verify-ci` because verfiy-ci can run dm and cdc tests concurrently and count the coverage seprately

### What is changed and how it works?

* add `/run-verify-ci` in command doc
* remove x86 linux check and build in github action
* add dm-docker build in github action

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ parsers:
       macro: no
 
 comment:
-  layout: "header, diff"
+  layout: "header, flags, diff"
```
Comment (Ehco1996) [inline .codecov.yml:27]: this is to distinguish different part of code
https://github.com/pingcap/ticdc/pull/3586#issuecomment-981556744

**Review by Ehco1996 — COMMENTED** [review]
seem it's ready to review

**Review by hi-rustin — APPROVED** [review]
LGTM

File: docs/ci/command.md [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 | ci pipeline                                 | Commands                                                    | Only triggered by command |
 | ------------------------------------------- | ----------------------------------------------------------- | ------------------------- |
-| idc-jenkins-ci-ticdc/unit-test              | /run-unit-test <br />/run-all-tests<br />/merge             | No                        |
+| jenkins-ticdc/verify                        | /run-verify-ci<br />/run-all-tests<br />/merge              | No                        |
```
Comment (hi-rustin) [inline docs/ci/command.md:15]: I understand that this task is still needed on other branches?

**Review by glorv — APPROVED** [review]

File: docs/ci/command.md (hunk 2 above)
Comment (Ehco1996) [reply docs/ci/command.md:15]: nice catch, i will add `unit-test` back in next commit

Comment (Ehco1996) [reply docs/ci/command.md:15]: @hi-rustin added in https://github.com/pingcap/ticdc/pull/3586/commits/202e5c09371cf84c8dabe3de76aa3a1937231559 do you have more suggestions? if not, i will merge this pr

**Review by hi-rustin — APPROVED** [review]
Thanks!

**Pull request merged**